src/BatchProcessor.ts

```diff
@@ -1,19 +1,245 @@
+    for(let key in msg) {
+      let type = msg[key].split('~').pop()
+      let value = (Inks as any).evaluate(msg[key], { out, ctx }, { sep: '~' })
+  
+      if(null != value && Types[type]) {
+        let validate = Gubu(eval(type))
+        validate(value)
+      }
+  
+      if(null != value) {
+        new_msg[key] = value
+      } else {
+        new_msg[key] = msg[key]
+      }
+  
+    }
+    
+    return new_msg
+  }
+  
+  static async workflowRun(seneca: any, msg: any, config: any, options: any, results: any) {
+    if(config.mode == null && 'async' == options.send.mode) {
+      try {
+        let result = await seneca.post(msg)
+        results.push(result)
+      } catch(error) {
+        // handle error
+        throw error
+      }
+    } else if(config.mode == null && 'sync' == options.send.mode) {
+      seneca.act(msg, function(this: any, err: any, result: any) {
+        if(null == err) {
+          results.push(result)
+        } else {
+          // handle error
+          throw err
+        }
+      })
+    } else if(config.mode == 'async') {
+      try {
+        let result = await seneca.post(msg)
+        results.push(result)
+      } catch(error) {
+        // handle error
+        throw error
+      }
+    } else if(config.mode == 'sync' ) {
+      seneca.act(msg, function(this: any, err: any, result: any) {
+        if(null == err) {
+          results.push(result)
+        } else {
+          // handle error
+          throw err
+        }
+      })
+    }
+        
+  }
+  
+}
+
+function humanify(when?: number, flags: {
+  parts?: boolean
+  terse?: boolean
+} = {}) {
+  const d = when ? new Date(when) : new Date()
+  const iso = d.toISOString()
+
+  if (flags.parts) {
+    let parts = iso.split(/[-:T.Z]/).map(s => +s)
+    let i = 0
+    let out: any = {
+      year: parts[i++],
+      month: parts[i++],
+      day: parts[i++],
+      hour: parts[i++],
+      minute: parts[i++],
+      second: parts[i++],
+      milli: parts[i++],
+    }
+    if (flags.terse) {
+      out = {
+        ty: out.year,
+        tm: out.month,
+        td: out.day,
+        th: out.hour,
+        tn: out.minute,
+        ts: out.second,
+        ti: out.milli,
+      }
+    }
+    return out
+  }
+
+  return +(iso.replace(/[^\d]/g, '').replace(/\d$/, ''))
+}
 
 function BatchProcessor(this: any, options: BatchProcessorOptionsFull) {
   const seneca: any = this
+  
+  
+  const Patrun = seneca.util.Patrun
+  const Jsonic = seneca.util.Jsonic
+  const Deep = seneca.util.deepextend
+  const generate_id = seneca.util.Nid
+  const BatchId = generate_id()
+  
+  let wheres: any = new Patrun({ gex: true })
+  
+  for(const message_whence in options.where) {
+    let match = options.where[message_whence]?.match
+    let pattern = Jsonic(message_whence)
+    for(const message_pattern in match) {
+      let workflow = match[message_pattern]
+      // console.log(message_pattern, pattern)
+      if(ALL == message_pattern) {
```

Comment:
check if the patrun pattern `''` (empty string) can be used as an ALL match - avoids need for special case code